velox/exec/Window.cpp

```diff
@@ -376,15 +548,38 @@ void Window::callApplyForPartitionRows(
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
 
     for (auto w = 0; w < numFuncs; w++) {
-      auto frameEndPoints = findFrameEndPoints(
-          w,
+      std::optional<vector_size_t> startOffset = std::nullopt;
```

Comment:
It might be better to determine the frameOffsets in the findFrameEndPointkPreceding(Following) functions, rather than introduce if statements here.

Reply:
Removed these checks, thanks!